packages/kbn-search-response-warnings/src/components/search_response_warnings/badge.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+import { EuiButton, EuiIcon, EuiPopover, useEuiTheme, useEuiFontSize } from '@elastic/eui';
+import { SearchResponseWarningsBadgePopoverContent } from './badge_popover_content';
+import type { SearchResponseWarning } from '../../types';
+
+interface Props {
+  visualizationLabel?: string;
+  warnings: SearchResponseWarning[];
+}
+
+export const SearchResponseWarningsBadge = (props: Props) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
+  const { euiTheme } = useEuiTheme();
+  const xsFontSize = useEuiFontSize('xs').fontSize;
+
+  if (!props.warnings.length) {
+    return null;
+  }
+
+  return (
+    <EuiPopover
```

Comment:
I think we had some styles for limiting the saved search popover's width. For this PR it looks like this:

Saved search:
<img width="300" alt="Screenshot 2023-10-26 at 14 54 20" src="https://github.com/elastic/kibana/assets/1415710/63c69002-c584-4700-a977-a758e88242c4">

Lens:
<img width="300" alt="Screenshot 2023-10-26 at 14 54 25" src="https://github.com/elastic/kibana/assets/1415710/70bc3975-fdae-4f4d-8cf4-55bbf89da7ef">

Can we make it look more like Len's popover?